Code/Framework/AzCore/AzCore/RTTI/BehaviorContext.h

```diff
@@ -2015,7 +2015,9 @@ namespace AZ
                 }
                 else
                 {
+                    AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
                     param.m_value = &const_cast<Type&>(result);
+                    AZ_POP_DISABLE_WARNING_GCC;
```

Comment:
I believe this dangling pointer can be fixed, first getting the address of the `result` variable (which in this case is guaranteed to not be a pointer type because this code is in the else block of an `if constexpr` statement).
Next that address can have it's constness casted of it.
```suggestion
                    AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
                    param.m_value = const_cast<Type*>(&result);
                    AZ_POP_DISABLE_WARNING_GCC;
```

Reply:
Well
```c++
param.m_value = const_cast<Type*>(&result);
```
still ends up as address of local variable :/

Reply:
everything else was was fixed properly, thanks!